README.MD
# Toast Notification 🍞

<p align="left">
  <a href="https://www.npmjs.com/package/toast-notification-js"><img src="https://img.shields.io/npm/v/toast-notification-js.svg" alt="NPM Release"></a>
  <a href="https://github.com/danie-ramdhani/toast-notification/releases"><img src="https://img.shields.io/github/release/danie-ramdhani/toast-notification?color=green" alt="Latest Release"></a>
</p>

A simple notification with better customization.

## Demo / Example

You can <a href="https://danie-ramdhani.github.io/toast-notification/">click here</a> to see the demo

## Installation

You can install it via NPM

```sh
npm i toast-notification-js
```

or you can include it on you page via CDN

on head tag:

```html
<link rel="stylesheet" href="https://unpkg.com/toast-notification-js/dist/toast-notification.min.css" />
```

on body tag:

```html
<script src="https://unpkg.com/toast-notification-js/dist/toast-notification.min.js"></script>
```

## ⏩ Quick Start:

```javascript
var toastNotification = ToastNotification();

toastNotification.create();

// or

toastNotification.create({
  classList: "your-favorite-custom-color",
  icon: "🍞",
  duration: "10",
  position: {
    y: "top",
  },
  // and other options...
});
```

## ⚙ Options

| Name                  | Type              | Description                                                                                                                                                                                                     | Default Value            |
| --------------------- | ----------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------ |
| classList             | String            | Your background color classes for the toast element, or `red` background if none are specified.                                                                                                                 | null                     |
| title                 | String            | The title to display on the toast element, or null if none is specified.                                                                                                                                        | null                     |
| message               | String            | The message to display on the toast element. If none is specified, the default value will be used.                                                                                                              | "This is Toast message!" |
| icon                  | String            | The icon to display on the toast element, or `null` if none is specified. You can input HTML String in this option.                                                                                             | null                     |
| duration              | Number            | The amount of time in seconds that the toast will be displayed. If duration is less than or equal to 0, the toast will permanently displayed. If `duration` is not initialized then default value will be used. | 5 seconds                |
| closeButton           | Boolean           | Whether to display a close button on the toast element.                                                                                                                                                         | true                     |
| position.x            | String            | The horizontal position of the toast element on the screen. Can be `"left"`, `"center"`, or `"right"`. If `position.x` is not specified, the default value will be used.                                        | "right"                  |
| position.y            | String            | The vertical position of the toast element on the screen. Can be `"top"`, or `"bottom"`. If `position.y` is not specified, the default value will be used.                                                      | "bottom"                 |
| position.z            | Number            | The z-index of the toast element. If `position.z` is not specified, the default value will be used.                                                                                                             | 50                       |
| redirect              | String or Object  | You can pass url string here or you can pass object with key `url` and `newWindow`. if the value is url string, the link will automatically open the url in new window.                                         | null                     |
| redirect.url          | String            | The URL to redirect to when the toast element is clicked.                                                                                                                                                       | null                     |
| redirect.newWindow    | String            | Whether to open the redirect URL in a new window or in the current one.                                                                                                                                         | true                     |
| progressBar           | Boolean or Object | You can pass `false` or `0` value to disable progress bar. or you can pass object with key `show` and `classList`.                                                                                              | true                     |
| progressBar.show      | Boolean           | Whether to display a progress bar on the toast element.                                                                                                                                                         | true                     |
| progressBar.classList | String            | Your background color classes for the progress bar color, or `currentColor` if none is specified.                                                                                                               | null                     |


CHANGELOG.MD
# Changelog

## [1.0.0] - 2023-04-22

- Initial Release

## [1.0.1] - 2023-04-27

### Changed:

- change duration option

### fixed:

- centering `toast-icon` & `toast-close-button` class using `flex`

## [1.0.2] - 2023-05-05

### fixed:

- fix notification stay in same place

## [1.0.2] - 2023-05-07

### fixed:

- fix duration option on `0` value


src/toast-message.js
function ToastMessage() {
  let allElements = {
    top: [],
    bottom: [],
  };

  let toastMessage = {};

  toastMessage.create = (preferences = {}) => {
    const options = {
      classList: preferences.classList ? preferences.classList : null,
      title: preferences.title ? preferences.title : null,
      message: preferences.message || "This is Toast message!",
      icon: preferences.icon ? preferences.icon : null,
      duration: preferences.duration && preferences.duration > 0 ? preferences.duration * 1000 : preferences.duration <= 0 ? -1 : 5000,
      closeButton: ["false", "0"].includes(String(preferences.closeButton).toLowerCase()) ? false : true,
      position: {
        x: preferences.position && ["left", "center", "right"].includes(preferences.position.x) ? preferences.position.x : "right",
        y: preferences.position && ["bottom", "top"].includes(preferences.position.y) ? preferences.position.y : "bottom",
        z: preferences.position && preferences.position.z ? preferences.position.z : 50,
      },
      redirect:
        typeof preferences.redirect == "string"
          ? preferences.redirect
          : {
              url: preferences.redirect && preferences.redirect.url ? preferences.redirect.url : null,
              newWindow:
                preferences.redirect && ["false", "0"].includes(String(preferences.redirect.newWindow).toLowerCase()) ? false : true,
            },
      progressBar: ["false", "0"].includes(String(preferences.progressBar).toLowerCase())
        ? false
        : {
            show: preferences.progressBar && ["false", "0"].includes(String(preferences.progressBar.show).toLowerCase()) ? false : true,
            classList: preferences.progressBar && preferences.progressBar.classList ? preferences.progressBar.classList : null,
          },
    };

    const offset = 10;
    const transitionDuration = 300;

    var icon = null;
    var title = null;
    const message = document.createElement(typeof options.redirect == "string" || options.redirect.url ? "a" : "div");
    const content = document.createElement("div");
    var closeButton = null;
    var progressBar = null;
    const toastBody = document.createElement("div");
    const toastWrapper = document.createElement("div");

    // toast
    toastBody.className += "toast ";
    if (options.classList) {
      toastBody.className += options.classList;
    } else {
      toastBody.style.backgroundColor = window.matchMedia("(prefers-color-scheme: dark)").matches ? "#f43f5e" : "#e11d48";
      toastBody.style.color = "#ffffff";
    }
    // icon
    if (options.icon) {
      icon = document.createElement("div");
      var iconContent = document.createElement("div");
      iconContent.classList.add("toast-icon");
      iconContent.innerHTML = options.icon;
      icon.append(iconContent);
      toastBody.append(icon);
    }

    // main content
    // toast title
    if (options.title) {
      title = document.createElement("div");
      title.classList.add("toast-title");
      title.append(options.title);
      content.append(title);
    }

    // toast message
    if (typeof options.redirect == "string") {
      message.href = options.redirect;
      message.style.textDecorationLine = "underline";
      message.style.textUnderlineOffset = "2px";
      message.target = "_blank";
      message.rel = "noopener noreferrer";
    } else if (options.redirect.url) {
      message.href = options.redirect.url;
      message.style.textDecorationLine = "underline";
      message.style.textUnderlineOffset = "2px";
      if (options.redirect.newWindow) {
        message.target = "_blank";
        message.rel = "noopener noreferrer";
      }
    }
    message.classList.add("toast-message");
    message.append(options.message);

    // toast content wrapper (wrapper for title and message)
    content.classList.add("toast-content-wrapper");
    content.append(message);

    toastBody.append(content);
    // end main content

    // close button
    if (options.closeButton) {
      closeButton = document.createElement("button");
      closeButton.type = "button";
      closeButton.classList.add("toast-close-button");
      closeButton.innerHTML = "&#10006;";
      toastBody.append(closeButton);
    }

    // progress bar
    if (options.progressBar.show && options.duration >= 0) {
      progressBar = document.createElement("div");
      progressBar.className += "progress-bar ";
      options.progressBar.classList
        ? (progressBar.className += options.progressBar.classList)
        : (progressBar.style.backgroundColor = "currentColor");
      progressBar.style.animationDuration = options.duration + "ms";
      toastBody.append(progressBar);
    }

    // toast wrapper
    toastWrapper.classList.add("toast-wrapper");
    toastWrapper.classList.add(options.position.x);
    toastWrapper.style.zIndex = options.position.z; // z-index
    toastWrapper.style.transitionProperty = "all";
    toastWrapper.style.transitionTimingFunction = "cubic-bezier(0.4, 0, 0.2, 1)";
    toastWrapper.style.transitionDuration = transitionDuration + "ms";
    toastWrapper.appendChild(toastBody);

    let toastMainFunction = {};

    toastMainFunction.createElement = () => {
      // insert element below body
      const newElement = document.body.insertAdjacentElement("afterbegin", toastWrapper);

      // close button on click
      if (options.closeButton) {
        closeButton.addEventListener("click", (event) => {
          event.stopPropagation();
          toastMainFunction.removeElement(newElement);
        });
      }

      // new element transition
      const newElementOffset =
        allElements[options.position.y][allElements[options.position.y].length - 1] !== undefined
          ? parseFloat(allElements[options.position.y][allElements[options.position.y].length - 1].style[options.position.y])
          : offset;
      toastWrapper.style[options.position.y] = newElementOffset + "px";

      // push new element
      allElements[options.position.y].push(newElement);

      // adjust position of previous elements
      allElements[options.position.y].slice(0, allElements[options.position.y].length - 1).forEach((element) => {
        const currentPos = parseFloat(element.style[options.position.y]);
        element.style[options.position.y] = `${currentPos + newElement.offsetHeight + offset}px`;
      });

      // auto remove function
      toastMainFunction.autoRemove(newElement);
    };

    toastMainFunction.removeElement = (element) => {
      const index = allElements[options.position.y].indexOf(element);
      const removedHeight = element.offsetHeight;

      // opacity transition
      element.style.opacity = "0";

      // Remove current element from allElements
      allElements[options.position.y].splice(index, 1);

      // Delaying remove current element from DOM for opacity transition effect
      setTimeout(() => {
        // remove element from DOM
        element.remove();

        // adjust position of previous elements
        allElements[options.position.y].slice(0, index).forEach((toastElement) => {
          const currentPos = parseFloat(toastElement.style[options.position.y] || 0);
          toastElement.style[options.position.y] = `${currentPos - removedHeight - offset}px`;
        });
      }, transitionDuration / 2);
    };

    toastMainFunction.autoRemove = (element) => {
      // remove element using duration
      if (options.duration >= 0 && !options.progressBar.show) {
        setTimeout(() => {
          toastMainFunction.removeElement(element);
        }, options.duration);
      }
      // remove element using progress bar
      if (options.duration >= 0 && options.progressBar.show) {
        progressBar.addEventListener("animationend", (event) => {
          event.stopPropagation();

          progressBar.style.opacity = 0;

          toastMainFunction.removeElement(element);
        });
      }
    };

    return toastMainFunction.createElement();
  };

  return toastMessage;
}


src/toast-notification.js
function ToastNotification() {
  let allElements = {
    top: [],
    bottom: [],
  };

  let toastMessage = {};

  toastMessage.create = (preferences = {}) => {
    const options = {
      classList: preferences.classList ? preferences.classList : null,
      title: preferences.title ? preferences.title : null,
      message: preferences.message || "This is Toast message!",
      icon: preferences.icon ? preferences.icon : null,
      duration: !preferences.duration ? 5000 : preferences.duration * 1000,
      closeButton: ["false", "0"].includes(String(preferences.closeButton).toLowerCase()) ? false : true,
      position: {
        x: preferences.position && ["left", "center", "right"].includes(preferences.position.x) ? preferences.position.x : "right",
        y: preferences.position && ["bottom", "top"].includes(preferences.position.y) ? preferences.position.y : "bottom",
        z: preferences.position && preferences.position.z ? preferences.position.z : 50,
      },
      redirect:
        preferences.redirect && typeof preferences.redirect == "string"
          ? preferences.redirect
          : {
              url: preferences.redirect && preferences.redirect.url ? preferences.redirect.url : null,
              newWindow: preferences.redirect && ["false", "0"].includes(String(preferences.redirect.newWindow).toLowerCase()) ? false : true,
            },
      progressBar: ["false", "0"].includes(String(preferences.progressBar).toLowerCase())
        ? false
        : {
            show: preferences.progressBar && ["false", "0"].includes(String(preferences.progressBar.show).toLowerCase()) ? false : true,
            classList: preferences.progressBar && preferences.progressBar.classList ? preferences.progressBar.classList : null,
          },
    };

    const offset = 10;
    const transitionDuration = 300;

    var icon = null;
    var title = null;
    const message = document.createElement(typeof options.redirect == "string" || options.redirect.url ? "a" : "div");
    const content = document.createElement("div");
    var closeButton = null;
    var progressBar = null;
    const toastBody = document.createElement("div");
    const toastWrapper = document.createElement("div");

    // toast
    toastBody.className += "toast ";
    if (options.classList) {
      toastBody.className += options.classList;
    } else {
      toastBody.style.backgroundColor = window.matchMedia("(prefers-color-scheme: dark)").matches ? "#f43f5e" : "#e11d48";
      toastBody.style.color = "#ffffff";
    }
    // icon
    if (options.icon) {
      icon = document.createElement("div");
      var iconContent = document.createElement("div");
      iconContent.classList.add("toast-icon");
      iconContent.innerHTML = options.icon;
      icon.append(iconContent);
      toastBody.append(icon);
    }

    // main content
    // toast title
    if (options.title) {
      title = document.createElement("div");
      title.classList.add("toast-title");
      title.append(options.title);
      content.append(title);
    }

    // toast message
    if (typeof options.redirect == "string") {
      message.href = options.redirect;
      message.style.color = "currentColor";
      message.style.textDecorationLine = "underline";
      message.style.textUnderlineOffset = "2px";
      message.target = "_blank";
      message.rel = "noopener noreferrer";
    } else if (options.redirect.url) {
      message.href = options.redirect.url;
      message.style.color = "currentColor";
      message.style.textDecorationLine = "underline";
      message.style.textUnderlineOffset = "2px";
      if (options.redirect.newWindow) {
        message.target = "_blank";
        message.rel = "noopener noreferrer";
      }
    }
    message.classList.add("toast-message");
    message.append(options.message);

    // toast content wrapper (wrapper for title and message)
    content.classList.add("toast-content-wrapper");
    content.append(message);

    toastBody.append(content);
    // end main content

    // close button
    if (options.closeButton) {
      closeButton = document.createElement("button");
      closeButton.type = "button";
      closeButton.classList.add("toast-close-button");
      closeButton.innerHTML = "&#10006;";
      toastBody.append(closeButton);
    }

    // progress bar
    if (options.progressBar.show && options.duration > 0) {
      progressBar = document.createElement("div");
      progressBar.className += "progress-bar ";
      options.progressBar.classList ? (progressBar.className += options.progressBar.classList) : (progressBar.style.backgroundColor = "currentColor");
      progressBar.style.animationDuration = options.duration + "ms";
      toastBody.append(progressBar);
    }

    // toast wrapper
    toastWrapper.classList.add("toast-wrapper");
    toastWrapper.classList.add(options.position.x);
    toastWrapper.style.zIndex = options.position.z; // z-index
    toastWrapper.style.transitionProperty = "all";
    toastWrapper.style.transitionTimingFunction = "cubic-bezier(0.4, 0, 0.2, 1)";
    toastWrapper.style.transitionDuration = transitionDuration + "ms";
    toastWrapper.appendChild(toastBody);

    let toastMainFunction = {};

    toastMainFunction.createElement = () => {
      // insert element below body
      const newElement = document.body.insertAdjacentElement("afterbegin", toastWrapper);

      // close button on click
      if (options.closeButton) {
        closeButton.addEventListener("click", (event) => {
          event.stopPropagation();
          toastMainFunction.removeElement(newElement);
        });
      }

      // new element transition
      const newElementOffset =
        allElements[options.position.y][allElements[options.position.y].length - 1] !== undefined
          ? parseFloat(allElements[options.position.y][allElements[options.position.y].length - 1].style[options.position.y])
          : offset;
      toastWrapper.style[options.position.y] = newElementOffset + "px";

      // push new element
      allElements[options.position.y].push(newElement);

      // adjust position of previous elements
      allElements[options.position.y].slice(0, allElements[options.position.y].length - 1).forEach((element) => {
        const currentPos = parseFloat(element.style[options.position.y]);
        element.style[options.position.y] = `${currentPos + newElement.offsetHeight + offset}px`;
      });

      // auto remove function
      toastMainFunction.autoRemove(newElement);
    };

    toastMainFunction.removeElement = (element) => {
      const index = allElements[options.position.y].indexOf(element);
      const removedHeight = element.offsetHeight;

      // opacity transition
      element.style.opacity = "0";

      // Remove current element from allElements
      allElements[options.position.y].splice(index, 1);

      // Delaying remove current element from DOM for opacity transition effect
      setTimeout(() => {
        // remove element from DOM
        element.remove();

        // adjust position of previous elements
        allElements[options.position.y].slice(0, index).forEach((toastElement) => {
          const currentPos = parseFloat(toastElement.style[options.position.y] || 0);
          toastElement.style[options.position.y] = `${currentPos - removedHeight - offset}px`;
        });
      }, transitionDuration / 2);
    };

    toastMainFunction.autoRemove = (element) => {
      // remove element using duration
      if (options.duration > 0 && !options.progressBar.show) {
        setTimeout(() => {
          toastMainFunction.removeElement(element);
        }, options.duration);
      }
      // remove element using progress bar
      if (options.duration > 0 && options.progressBar.show) {
        progressBar.addEventListener("animationend", (event) => {
          event.stopPropagation();

          progressBar.style.opacity = 0;

          toastMainFunction.removeElement(element);
        });
      }
    };

    return toastMainFunction.createElement();
  };

  return toastMessage;
}


src/toast-notification.umd.js
((global, factory) => {
  typeof exports === "object" && typeof module !== "undefined"
    ? (module.exports = factory())
    : typeof define === "function" && define.amd
    ? define(factory)
    : ((global = typeof globalThis !== "undefined" ? globalThis : global || self), (global.ToastNotification = factory()));
})(this, () => {
  "use strict";

  let toastNotificationElements = {
    top: [],
    bottom: [],
  };

  return () => {
    ToastNotification.create = (preferences = {}) => {
      const options = {
        classList: preferences.classList ? preferences.classList : null,
        title: preferences.title ? preferences.title : null,
        message: preferences.message || "This is Toast message!",
        icon: preferences.icon ? preferences.icon : null,
        duration: preferences.duration ? preferences.duration * 1000 : !preferences.duration && preferences.duration <= 0 ? 0 : 5000,
        closeButton: ["false", "0"].includes(String(preferences.closeButton).toLowerCase()) ? false : true,
        position: {
          x: preferences.position && ["left", "center", "right"].includes(preferences.position.x) ? preferences.position.x : "right",
          y: preferences.position && ["bottom", "top"].includes(preferences.position.y) ? preferences.position.y : "bottom",
          z: preferences.position && preferences.position.z ? preferences.position.z : 50,
        },
        redirect:
          preferences.redirect && typeof preferences.redirect == "string"
            ? preferences.redirect
            : {
                url: preferences.redirect && preferences.redirect.url ? preferences.redirect.url : null,
                newWindow: preferences.redirect && ["false", "0"].includes(String(preferences.redirect.newWindow).toLowerCase()) ? false : true,
              },
        progressBar: ["false", "0"].includes(String(preferences.progressBar).toLowerCase())
          ? false
          : {
              show: preferences.progressBar && ["false", "0"].includes(String(preferences.progressBar.show).toLowerCase()) ? false : true,
              classList: preferences.progressBar && preferences.progressBar.classList ? preferences.progressBar.classList : null,
            },
      };

      const offset = 10;
      const transitionDuration = 300;

      var icon = null;
      var title = null;
      const message = document.createElement(typeof options.redirect == "string" || options.redirect.url ? "a" : "div");
      const content = document.createElement("div");
      var closeButton = null;
      var progressBar = null;
      const toastBody = document.createElement("div");
      const toastWrapper = document.createElement("div");

      // toast
      toastBody.className += "toast ";
      if (options.classList) {
        toastBody.className += options.classList;
      } else {
        toastBody.style.backgroundColor = window.matchMedia("(prefers-color-scheme: dark)").matches ? "#f43f5e" : "#e11d48";
        toastBody.style.color = "#ffffff";
      }
      // icon
      if (options.icon) {
        icon = document.createElement("div");
        var iconContent = document.createElement("div");
        iconContent.classList.add("toast-icon");
        iconContent.innerHTML = options.icon;
        icon.append(iconContent);
        toastBody.append(icon);
      }

      // main content
      // toast title
      if (options.title) {
        title = document.createElement("div");
        title.classList.add("toast-title");
        title.append(options.title);
        content.append(title);
      }

      // toast message
      if (typeof options.redirect == "string") {
        message.href = options.redirect;
        message.style.color = "currentColor";
        message.style.textDecorationLine = "underline";
        message.style.textUnderlineOffset = "2px";
        message.target = "_blank";
        message.rel = "noopener noreferrer";
      } else if (options.redirect.url) {
        message.href = options.redirect.url;
        message.style.color = "currentColor";
        message.style.textDecorationLine = "underline";
        message.style.textUnderlineOffset = "2px";
        if (options.redirect.newWindow) {
          message.target = "_blank";
          message.rel = "noopener noreferrer";
        }
      }
      message.classList.add("toast-message");
      message.append(options.message);

      // toast content wrapper (wrapper for title and message)
      content.classList.add("toast-content-wrapper");
      content.append(message);

      toastBody.append(content);
      // end main content

      // close button
      if (options.closeButton) {
        closeButton = document.createElement("button");
        closeButton.type = "button";
        closeButton.classList.add("toast-close-button");
        closeButton.innerHTML = "&#10006;";
        toastBody.append(closeButton);
      }

      // progress bar
      if (options.progressBar.show && options.duration > 0) {
        progressBar = document.createElement("div");
        progressBar.className += "progress-bar ";
        options.progressBar.classList
          ? (progressBar.className += options.progressBar.classList)
          : (progressBar.style.backgroundColor = "currentColor");
        progressBar.style.animationDuration = options.duration + "ms";
        toastBody.append(progressBar);
      }

      // toast wrapper
      toastWrapper.classList.add("toast-wrapper");
      toastWrapper.classList.add(options.position.x);
      toastWrapper.style.zIndex = options.position.z; // z-index
      toastWrapper.style.transitionProperty = "all";
      toastWrapper.style.transitionTimingFunction = "cubic-bezier(0.4, 0, 0.2, 1)";
      toastWrapper.style.transitionDuration = transitionDuration + "ms";
      toastWrapper.appendChild(toastBody);

      var createElement = () => {
        // insert element below body
        const newElement = document.body.insertAdjacentElement("afterbegin", toastWrapper);

        // close button on click
        if (options.closeButton) {
          closeButton.addEventListener("click", (event) => {
            event.stopPropagation();
            removeElement(newElement);
          });
        }

        // new element transition
        const newElementOffset =
          toastNotificationElements[options.position.y][toastNotificationElements[options.position.y].length - 1] !== undefined
            ? parseFloat(
                toastNotificationElements[options.position.y][toastNotificationElements[options.position.y].length - 1].style[options.position.y]
              )
            : offset;
        toastWrapper.style[options.position.y] = newElementOffset + "px";

        // push new element
        toastNotificationElements[options.position.y].push(newElement);

        // adjust position of previous elements
        toastNotificationElements[options.position.y].slice(0, toastNotificationElements[options.position.y].length - 1).forEach((element) => {
          const currentPos = parseFloat(element.style[options.position.y]);
          element.style[options.position.y] = `${currentPos + newElement.offsetHeight + offset}px`;
        });

        // auto remove function
        autoRemoveElement(newElement);
      };

      var removeElement = (element) => {
        const index = toastNotificationElements[options.position.y].indexOf(element);
        const removedHeight = element.offsetHeight;

        // opacity transition
        element.style.opacity = "0";

        // Remove current element from toastNotificationElements
        toastNotificationElements[options.position.y].splice(index, 1);

        // Delaying remove current element from DOM for opacity transition effect
        setTimeout(() => {
          // remove element from DOM
          element.remove();

          // adjust position of previous elements
          toastNotificationElements[options.position.y].slice(0, index).forEach((toastElement) => {
            const currentPos = parseFloat(toastElement.style[options.position.y] || 0);
            toastElement.style[options.position.y] = `${currentPos - removedHeight - offset}px`;
          });
        }, transitionDuration / 2);
      };

      var autoRemoveElement = (element) => {
        // remove element using duration
        if (options.duration > 0 && !options.progressBar.show) {
          setTimeout(() => {
            removeElement(element);
          }, options.duration);
        }
        // remove element using progress bar
        if (options.duration > 0 && options.progressBar.show) {
          progressBar.addEventListener("animationend", (event) => {
            event.stopPropagation();

            progressBar.style.opacity = 0;

            removeElement(element);
          });
        }
      };

      return createElement();
    };

    return ToastNotification;
  };
});
